core/trino-spi/src/main/java/io/trino/spi/block/SingleRowBlockWriter.java

```diff
@@ -74,14 +76,20 @@ protected Block getRawFieldBlock(int fieldIndex)
         return fieldBlockBuilders[fieldIndex];
     }
 
+    @Override
+    protected int getRowIndex()
+    {
+        return rowIndex;
+    }
+
     @Override
     public long getSizeInBytes()
     {
         long currentBlockBuilderSize = 0;
         for (BlockBuilder fieldBlockBuilder : fieldBlockBuilders) {
-            currentBlockBuilderSize += fieldBlockBuilder.getSizeInBytes();
+            currentBlockBuilderSize += fieldBlockBuilder.getSizeInBytes() - fieldBlockBuilder.getRegionSizeInBytes(0, rowIndex);
```

Comment:
why not just `fieldBlockBuilder.getRegionSizeInBytes(rowIndex, 1)`?

Reply:
Because it does not work for "empty" row? By empty I mean the `SingleRowBlockWriterInstance` after call `beginBlockEntry`. It should return 0 (as version before my change). 

This one `fieldBlockBuilder.getRegionSizeInBytes(rowIndex, 1)` returns non-zero.

Reply:
I don't understand, it should happen to be truth
```
fieldBlockBuilder.getRegionSizeInBytes(rowIndex, 1) = fieldBlockBuilder.getSizeInBytes() - fieldBlockBuilder.getRegionSizeInBytes(0, rowIndex)
```
If it isn't, then why it's the case?

Reply:
If you run test TestSingleRowBlockWriter with `fieldBlockBuilder.getRegionSizeInBytes(rowIndex, 1)`, it fails whileas the current and previous implementation passes the test. 
What about the case?:
```
rowBlockBuilder = (RowBlockBuilder) RowType.anonymous(types).createBlockBuilder(null, 8);
SingleRowBlockWriter singleRowBlockWriter = rowBlockBuilder.beginBlockEntry(); // assert the rowIndex == 0
fieldBlockBuilder.getRegionSizeInBytes(rowIndex, 1) != 0 whileas nothing was written to writer. 
```